src/mat.c

```diff
@@ -1171,7 +1192,7 @@ Mat_VarDelete(mat_t *mat, const char *name)
     if ( NULL == mat || NULL == name )
         return err;
 
-    if ( (tmp_name = mktemp(temp)) != NULL ) {
+    if ( (tmp_name = Mat_mktemp(temp)) != NULL ) {
```

Comment:
Does the template needs to be adpated from "XXXXXX" to "/tmp/tmpdir.XXXXXX"?

Reply:
I don't think the old implementation with mktemp() was putting the temporary files in /tmp - I think it was just adding them to the current working directory (unless I misread the code).

However, I went ahead and made this change, because it seems like a good idea to put temporary files in /tmp anyway.

I also added code to delete the temporary directory (in addition to deleting the temporary file), in case that was why you suggested putting it in /tmp.